chainercv/utils/iterator/apply_prediction_to_iterator.py

```diff
@@ -0,0 +1,111 @@
+from chainercv.utils.iterator.unzip import unzip
+
+
+def apply_prediction_to_iterator(predict, iterator, hook=None):
+    """Apply a prediction function/method to an iterator.
+
+    This function applies a prediction function/method to an iterator.
+    It assumes that the iterator returns a batch of images or
+    a tuple whose first element is a batch of images. In the case that
+    it returns a tuple, the rests are treated as ground truth
+    values.
+
+    >>> imgs = next(iterator)
+    >>> # imgs: [img]
+    or
+    >>> imgs, gt_vals0, gt_vals1 = next(iterator)
+    >>> # gt_vals0: [gt_val0]
+    >>> # gt_vals1: [gt_val1]
+
+    This function applys :func:`predict` to a batch of images and gets
+    predicted value(s). :func:`predict` should take a batch of images and
+    return a batch of prediction values
+    or a tuple of batches of prediction values.
+
+    >>> pred_vals0 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    or
+    >>> pred_vals0, pred_vals1 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    >>> # pred_vals1: [pred_val1]
+
+    Args:
+        predict: A callable which takes a batch of images and returns
+            prediction.
+        iterator (chainer.Iterator): An iterator. Each sample should have
+            an image as its first element. This image is passed to
+            :func:`predict` as an argument.
+            The rests are treated as ground truth values.
+        hook: A callable which is called after each iteration.
+            :obj:`imgs`, :obj:`pred_labels` and :obj:`gt_values` are passed as
+            arguments.
+            Note that these values do not contain data from the previous
+            iterations.
+
+    Returns:
+        An iterator and two tuples of iterators:
+        This function returns an iterator and two tuples of iterators:
+        :obj:`imgs`, :obj:`pred_values` and :obj:`gt_values`.
+        :obj:`imgs` is an iterator which returns an image.
```

Comment:
How about listing descriptions of `imgs` `pred_values` and `gt_values` with `* CONTENT` .